### ISSUE TITLE
Getting started documentation is confusing

### ISSUE THREAD

**Issue by jasw** [docker/swarm#804]
Hi, I am trying to follow this guide (http://docs.docker.com/swarm/) to setup swarm with two nodes. As I am relatively new to Docker and Swarm, I wrote down everything I tried here. Might be helpful to beginners like me and the documentation maintainer to see where people got stuck at. 

Here is what I have done. 

1.created 3 new Ubuntu instances (14.04 LTS) from AWS, install docker by following the instructions here (https://docs.docker.com/installation/ubuntulinux/)
Note, I added docker into the ubuntu usergroup as recommended for convenience.

Lets call them instance A, B and C. A meant to be used as the manager, B and C would be nodes.

2.On all 3 instances , do `docker pull swarm` (dont know if thats needed, maybe its only needed for the 'manager' node). 

3.On instance A, do `docker run --rm swarm create`. Succeeded.

4.On instance B, do `docker -H tcp://0.0.0.0:2375 -d`. This resulted in error: 

````
INFO[0000] +job serveapi(tcp://0.0.0.0:2375)
INFO[0000] Listening for HTTP on tcp (0.0.0.0:2375)
INFO[0000] /!\ DON'T BIND ON ANY IP ADDRESS WITHOUT setting -tlsverify IF YOU DON'T KNOW WHAT YOU'RE DOING /!\
FATA[0000] Shutting down daemon due to errors: pid file found, ensure docker is not running or delete /var/run/docker.pid

````

This is caused by docker -d is already running. So I did a 'sudo service stop docker' then run this commend again `docker -H tcp://0.0.0.0:2375 -d` , now I got this error:

````
INFO[0000] +job serveapi(tcp://0.0.0.0:2375)
INFO[0000] Listening for HTTP on tcp (0.0.0.0:2375)
INFO[0000] /!\ DON'T BIND ON ANY IP ADDRESS WITHOUT setting -tlsverify IF YOU DON'T KNOW WHAT YOU'RE DOING /!\
FATA[0000] Shutting down daemon due to errors: open /var/run/docker.pid: permission denied
````

It seems to be because of permission of that /var/run folder, so added sudo to the run command now it succeeded and running in daemon mode.

5.I did a `ctrl+z` and `bg` to put that daemon into background.
 

6.On the same instance (B), do `docker run -d swarm join --addr=<node_ip:2375> token://<cluster_id>
`. Now I got this error:

````
FATA[0000] Post http:///var/run/docker.sock/v1.18/containers/create: dial unix /var/run/docker.sock: no such file or directory. Are you trying to connect to a TLS-enabled daemon without TLS?
````

After doing this command: `export DOCKER_HOST="tcp://0.0.0.0:2375"`. The swarm join command worked.  **Worth adding this to the document for beginners like me?**

7.On instance A, do `docker run -d -p 8888:2375 swarm manage token://a5e663675dd0093d4f443cf633970cda &`.  succeeded. 

Do a `docker ps` to verify:

````
CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS              PORTS                    NAMES
12e4ca329b8c        swarm:latest        "/swarm manage token   7 minutes ago       Up 7 minutes        0.0.0.0:8888->2375/tcp   distracted_jang
````

8.Following the guide to do a `docker -H tcp://0.0.0.0:2375` got an error:

````
ubuntu@ip-xxxx:~$ docker -H tcp://0.0.0.0:2375 info
FATA[0000] Cannot connect to the Docker daemon. Is 'docker -d' running on this host?

````

I simply tried to change the port to 8888 (the port for the docker manager) and run again:

''''
ubuntu@ip-172-30-0-177:~$ docker -H tcp://0.0.0.0:8888 info
Containers: 0
Strategy: spread
Filters: affinity, health, constraint, port, dependency
Nodes: 0
''''

Kind of worked, but got 0 nodes. Shouldn't it be 1 (instance B)? So the question here would be  **is 8888 the right port?**

9.I read from other issues, this might be caused by the docker daemon not started on host mode. 
So , on instance A, I did 'sudo service docker stop' and 'docker -H tcp://0.0.0.0:2375 -d&'. 

Then I repeated step 7 command and step 8. Got the same result: 0 containers. 

**What did I do wrong?**

**Comment by jasw** [comment]
Just to add, doing a 'docker run --rm swarm list token://<cluster_id>' actually successfully reported my instance B and C's ip addresses as you can see here:

````

ubuntu@ip-172-30-0-177:~$ docker run --rm swarm list token://a5e663675dd0093d4f443cf633970cda
INFO[1677] POST /v1.18/containers/create
INFO[1677] +job create()
INFO[1677] +job log(create, 56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e, swarm:latest)
INFO[1677] -job log(create, 56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e, swarm:latest) = OK (0)
INFO[1677] -job create() = OK (0)
INFO[1677] POST /v1.18/containers/56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e/attach?stderr=1&stdout=1&stream=1
INFO[1677] +job container_inspect(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e)
INFO[1677] -job container_inspect(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e) = OK (0)
INFO[1677] +job attach(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e)
INFO[1677] POST /v1.18/containers/56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e/start
INFO[1677] +job start(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e)
INFO[1677] +job allocate_interface(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e)
INFO[1677] -job allocate_interface(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e) = OK (0)
INFO[1677] +job log(start, 56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e, swarm:latest)
INFO[1677] -job log(start, 56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e, swarm:latest) = OK (0)
INFO[1677] -job start(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e) = OK (0)
52.5.234.214:2375
52.7.98.126:2375
INFO[1677] +job log(die, 56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e, swarm:latest)
INFO[1677] -job log(die, 56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e, swarm:latest) = OK (0)
INFO[1677] -job attach(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e) = OK (0)
INFO[1677] +job release_interface(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e)
INFO[1677] -job release_interface(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e) = OK (0)
INFO[1677] POST /v1.18/containers/56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e/wait
INFO[1677] +job wait(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e)
INFO[1677] -job wait(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e) = OK (0)
INFO[1677] GET /v1.18/containers/56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e/json
INFO[1677] +job container_inspect(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e)
INFO[1677] -job container_inspect(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e) = OK (0)
INFO[1677] DELETE /v1.18/containers/56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e?v=1
INFO[1677] +job rm(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e)
INFO[1677] +job log(destroy, 56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e, swarm:latest)
INFO[1677] -job log(destroy, 56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e, swarm:latest) = OK (0)
INFO[1677] -job rm(56fb049ce6fe3507e7cfe86566ceda58e3573d5b0b7dbb23dcc07c5fd9684b0e) = OK (0)

````

So, why docker -H tcp://0.0.0.0:<manager_port> info does not report the right information?

**Comment by jasw** [comment]
I just changed the security group by allowing access to the tcp 2375 ports in the agents, now all worked. 
I guess I will need to read more about the TLS to make it secure, now its wide open to the world.

**Comment by aluzzardi** [comment]
Thank you @jasw, this is really helpful.

I'm going to re-open this issue and flag it as a documentation problem - it should be way more obvious.

**Comment by aluzzardi** [comment]
ping @moxiegirl

**Comment by moxiegirl** [comment]
@aluzzardi I'm working the updates on the new getting started with machine. Then, I planned to do the manual one. If you have additions to the manual one in the mean time, please make 'em. I'll pile mine on after.

**Comment by guiltry** [comment]
hi @jasw, I write up how to install docker swarm, I hope this article can help you. :smile: 
http://blog.guiltry.com/how-to-setting-docker-swarm-properly/

**Comment by scrisenbery** [comment]
@moxiegirl Along a similar note, I am seeing an error similar to that in @jasw 's step 6, however I have added the -H flag to the daemon and it still does not work. The manual documentation is not clear here http://docs.docker.com/swarm/install-manual/#create-swarm-nodes), specifically under step 2 of that section. I am not sure which machine that command is supposed to be run on. It is failing when I use the node I was configuring but it seems to work on the machine that is the master. Also, does <node_IP:2375> refer to the node you are configuring?

**Comment by MHBauer** [comment]
@scrisenbery in theory you run it on the node itself. This guarantees it has docker, and that if it goes down, it will stop advertising.

`swarm join` with the token discovery service can technically be run from anywhere, as long as it can access the internet. The manager also has to contact the internet.

**Comment by scrisenbery** [comment]
So if node1 is the master and node2 is the one I am trying to join to the swarm, I should run:
`docker run -d warm join --addr=node2:2375 token://token` ? And that could be run on wither node1 or node2? @MHBauer 

Both nodes have docker but I get the error mentioned in step 6 above if I run it on node2.

**Comment by aluzzardi** [comment]
@scrisenbery Hey. You should run the `join` on node2.

Basically, the join advertises to the cluster that the node is alive. If `node2` goes down, then `swarm join` will go down with it and eventually it will get removed from the node list.

In your cluster you should have:
- **1** `swarm manager` (you can have more with master/slave replication, but for the sake of simplicity let's assume only one manager)
- **N** `swarm join`, where N is the number of nodes in the cluster. `join` should run alongside the docker engine you want to deploy on

/cc @moxiegirl

**Comment by MHBauer** [comment]
I think I see the problem in the docs.
jasw wrote a potential solution.
Since the docs say to manually start with `-H 0.0.0.0:2375` the local socket isn't available.

@scrisenbery two things to try
 1. when starting on each node, run `docker daemon -D -H unix:///var/run/docker.sock -H 0.0.0.0:2375`
 2. a more immediate fix is to run `docker -H 0.0.0.0:2375 run -d warm join --addr=node2:2375 token://token` which says to look for the local server running on http.
